### PR TITLE
MEP_Engine: Added support to CameraDevice objects

### LINKED ISSUE #2147: MEP_Engine: Give support for Camera devices objects
#### Description:
Need to give support to some aspects of the newly created Camera object in MEP_oM

### PR DESCRIPTION
**Pull request by tiagogrossi**
### NOTE: Depends on 
[BHoM/BHoM#1085](https://github.com/BHoM/BHoM/pull/1085)

   
### Issues addressed by this PR

Closes #2147 

Added support for Camera device object

### Test files
Test file [here](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM/MEP_oM/Issue1078-Create%20definition%20for%20Camera%20devices/MEP_oM-Issue1078-Create%20definition%20for%20Camera%20devices.gh?csf=1&web=1&e=CERKJ1).

### Changelog
Added `BH.Engine.MEP.Compute.Error`
Modified `BH.Engine.MEP.Query.Geometry`
Modified `BH.Engine.MEP.Query.Centreline`

### CODE REVIEW

**Comment by tiagogrossi** [comment]
@FraserGreenroyd thanks for the advice on the geometry extraction for this one. After a chat with Isak I felt that `IImmutable` isn't suitable for this object because it has to be flexible. The geometry is driven by the first 3 properties of the object, that can change freely, so I'm not giving the option for the user to set the geometry randomly. Does that makes sense? Suggestion?

File: MEP_Engine/Compute/Error.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            Reflection.Compute.RecordWarning("The method works only with Polyline, since it needs to a closed perimeter.");
+        }
+
+        /******************************************/
+        
+        internal static void NonClosedElementError()
+        {
+            Reflection.Compute.RecordWarning("The method works only with closed curves Polyline.");
+        }
```
Comment (FraserGreenroyd) [inline MEP_Engine/Compute/Error.cs:48]: I don't like this kind of approach to this, and this isn't even being used anywhere in the rest of the code. Please delete it.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by enarhi** [comment]
Migrated to Security_Engine namespace as per https://github.com/BHoM/BHoM/pull/1091

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix project file ref. 1414955338

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
LGTM, thanks @enarhi for picking it up and moving to Security Engine - merging to unblock you and allowing you to continue on with @tiagogrossi :smile:

**Pull request merged**